src/stock-items/add-stock-item/packaging-units/packaging-units.component.tsx

```diff
@@ -213,43 +234,73 @@ const PackagingUnitRow: React.FC<{
   };
 
   return (
-    <TableRow>
-      <TableCell>
-        {isEditing ? (
-          <PackagingUnitsConceptSelector
-            row={row}
-            controllerName={"packagingUomUuid"}
-            name="packagingUomUuid"
-            control={control}
-            invalid={!!errors.packagingUomUuid}
-          />
-        ) : (
-          (!isEditing || !row.uuid.startsWith("new-item")) &&
-          row?.packagingUomName
-        )}
-      </TableCell>
-      <TableCell>
-        <div className={styles.packingTableCell}>
-          <ControlledNumberInput
-            row={row}
-            controllerName="factor"
-            name="factor"
-            control={control}
-            id={`${row.uuid}-${key}`}
-            invalid={!!errors.factor}
-          />
+    <>
+      <TableRow>
+        <TableCell>
+          {isEditing ? (
+            <PackagingUnitsConceptSelector
+              row={row}
+              controllerName={"packagingUomUuid"}
+              name="packagingUomUuid"
+              control={control}
+              invalid={!!errors.packagingUomUuid}
+            />
+          ) : (
+            (!isEditing || !row.uuid.startsWith("new-item")) &&
+            row?.packagingUomName
+          )}
+        </TableCell>
+        <TableCell>
+          <div className={styles.packingTableCell}>
+            <ControlledNumberInput
+              row={row}
+              controllerName="factor"
+              name="factor"
+              control={control}
+              id={`${row.uuid}-${key}`}
+              invalid={!!errors.factor}
+            />
 
-          <Button
-            type="button"
-            size="sm"
-            className="submitButton clear-padding-margin"
-            iconDescription={"Delete"}
-            kind="ghost"
-            renderIcon={TrashCan}
-            onClick={(e) => handleDelete(e)}
+            <Button
+              type="button"
+              size="sm"
+              className="submitButton clear-padding-margin"
+              iconDescription={"Delete"}
+              kind="ghost"
+              renderIcon={TrashCan}
+              onClick={() => launchDeleteModal(row.packagingUomName)}
+            />
+          </div>
+        </TableCell>
+      </TableRow>
+      {showModal && (
```

Comment:
stick to o3 implementation of dialog and modals

Reply:
> stick to o3 implementation of dialog and modals

This has been addressed too in the subsequent commit